src/core/api/paletteService.tsx

```diff
@@ -0,0 +1,54 @@
+/**
+ * ---------------------------------------------------------------------
+ * Copyright (c) 2020 EclipseSource Munich
+ * Licensed under MIT
+ * https://github.com/eclipsesource/jsonforms-editor/blob/master/LICENSE
+ * ---------------------------------------------------------------------
+ */
+
+import React from 'react';
+
+import { GroupIcon, HorizontalIcon, VerticalIcon } from '../icons';
+import { EditorUISchemaElement } from '../model/uischema';
+import { createLayout } from '../util/generators/uiSchema';
+
+export interface PaletteService {
+  getPaletteElements(): Map<string, PaletteElement[]>;
+}
+
+export interface PaletteElement {
+  type: string;
+  label: string;
+  icon: React.ReactNode;
+  uiSchemaElement: EditorUISchemaElement;
+}
+
+const paletteElements: Map<string, PaletteElement[]> = new Map([
+  [
+    'Layouts',
+    [
+      {
+        type: 'HorizontalLayout',
+        label: 'Horizontal Layout',
+        icon: <HorizontalIcon />,
+        uiSchemaElement: createLayout('HorizontalLayout'),
```

Comment:
I think this should be a factory instead of a concrete element. The problem here is that this element has a `uuid` which will be reused all the time if I'm not mistaken.

Reply:
True, good catch!